### PR TITLE
test: adds tests for removal of data values and attributes

### PR DESCRIPTION
**Pull request by vilkg**
- adds some tests to cover data value and attribute removal: https://jira.dhis2.org/browse/DHIS2-10797
- adds few tests to cover field filtering: https://jira.dhis2.org/browse/DHIS2-10756

### CODE REVIEW

**Review by gnespolino — APPROVED** [review]
Good job, looks good

File: dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/ProgramActions.java [hunk 1]
```diff
@@ -59,6 +59,9 @@ public ApiResponse createProgram( String programType )
     {
         JsonObject object = getDummy( programType );
 
+        if (programType.equalsIgnoreCase( "with_registration" )) {
```
Comment (larshelge) [inline dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/ProgramActions.java:62]: Codestyle, space after ( and before ).

We should probably apply the speedy spotless plugin to the e2e project as well.

Comment (vilkg) [reply dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/ProgramActions.java]: Thanks, fixed! Good point about speedy spotless  👍

File: dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/ProgramActions.java [hunk 2]
```diff
@@ -59,6 +59,9 @@ public ApiResponse createProgram( String programType )
...
+            object.addProperty( "trackedEntityType", "Q9GufDoplCL" );
+        }
         return post( object );
```
[the remark this answers is not in the document]
Comment (vilkg) [reply dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/ProgramActions.java]: yeah, thanks for noticing that! Fixed

**Review by lucaCambi77 — APPROVED** [review]

**Pull request merged**